pymesync.py

```diff
@@ -512,6 +515,42 @@ def delete_user(self, username=None):
 
         return self.__delete_object("users", username)
 
+    def token_expiration_time(self):
+        """
+        token_expiration_time()
+
+        Returns the expiration time of the JWT (JSON Web Token) associated with
+        this object.
+        """
+        # Check that user has authenticated
+        local_auth_error = self.__local_auth_error()
+        if local_auth_error:
+            return [{self.error: local_auth_error}]
+
+        # Return valid date if in test mode
+        if self.test:
+            return mock_pymesync.token_expiration_time()
+
+        # Decode the token, then get the second dict (payload) from the
+        # resulting string. The payload contains the expiration time.
+        try:
+            decoded_payload = base64.b64decode(self.token).split("}", 1)[1]
```

Comment:
The three portions of the token are separated by periods, *after* base64 encoding. The more robust way to do this would be:

``decoded_payload = base64.b64decode(self.token.split(".", 1)[1])``